rules/rules.js

```diff
@@ -371,85 +371,104 @@ const SwitchableJSRule = function (ruleConfig) {
-  let evArr = [];
-  if (ev.includes('triggered')) {
-    const atmp = ev.split(' triggered '); // astro:sun:local:astroDawn#event triggered START
-    evArr = [atmp[0], 'triggered', atmp[1]];
-  } else {
-    evArr = ev.split("'").join('').split('Item ').join('').split(' '); // Item 'benqth681_switch' received command ON
+  // Item triggers
+  if (input.containsKey('command')) {
+    data.receivedCommand = input.get('command').toString();
+    data.command = input.get('command').toString();
+  }
+  addFromHashMap(input, 'oldState', data);
+  addFromHashMap(input, 'newState', data);
+  if (input.containsKey('state')) {
+    data.receivedState = input.get('state').toString();
+    data.state = input.get('state').toString();
   }
 
-  const d = {
-    // size: input.size(),
-    oldState: input.get('oldState') + '',
-    newState: input.get('newState') + '',
-    state: input.get('state') + '', // this occurs on an ItemStateUpdateTrigger
-    receivedCommand: null,
-    receivedState: null,
-    receivedTrigger: null,
-    itemName: evArr[0],
-    module: input.get('module'),
-    command: input.get('command') + '' // for PWM trigger
-  };
-
-  try {
-    if (event !== null && event.getPayload()) {
-      d.payload = JSON.parse(event.getPayload());
-      log.debug('Extracted event payload {}', d.payload);
+  // Thing triggers
+  addFromHashMap(input, 'oldStatus', data);
+  addFromHashMap(input, 'newStatus', data);
+  addFromHashMap(input, 'status', data);
+
+  // Only with event data (for Item, Thing & Channel triggers)
+  if (event) {
+    switch (Java.typeName(event.class)) {
+      case 'org.openhab.core.items.events.ItemCommandEvent':
+        data.itemName = event.getItemName();
+        data.eventType = 'command';
+        data.triggerType = 'ItemCommandTrigger';
+        break;
+      case 'org.openhab.core.items.events.ItemStateChangedEvent':
+        data.itemName = event.getItemName();
+        data.eventType = 'change';
+        data.triggerType = 'ItemStateChangeTrigger';
+        break;
+      case 'org.openhab.core.items.events.ItemStateEvent':
+        data.itemName = event.getItemName();
+        data.eventType = 'update';
+        data.triggerType = 'ItemStateUpdateTrigger';
+        break;
+      case 'org.openhab.core.thing.events.ThingStatusInfoChangedEvent':
+        data.thingUID = event.getThingUID().toString();
+        data.eventType = 'change';
+        data.triggerType = 'ThingStatusChangeTrigger';
+        break;
+      case 'org.openhab.core.thing.events.ThingStatusInfoEvent':
+        data.thingUID = event.getThingUID().toString();
+        data.eventType = 'update';
+        data.triggerType = 'ThingStatusUpdateTrigger';
+        break;
+      case 'org.openhab.core.thing.events.ChannelTriggeredEvent':
+        data.channelUID = event.getChannel().toString();
+        data.receivedEvent = event.getEvent();
+        data.eventType = 'triggered';
+        data.triggerType = 'ChannelEventTrigger';
```

Comment:
```suggestion
        data.channelUID = event.getChannel().toString();
        data.receivedEvent = event.getEvent();
        data.eventType = 'triggered';
        data.triggerType = 'ChannelEventTrigger';
         Object.defineProperty( 
            data, 
            'receivedTrigger', 
            { 
                get: function() {
                    console.warn('receivedTrigger has been deprecated and will be removed in a future release.  Please use "receivedEvent" instead.')
                    return event.getEvent();
                }
            }
        )
```

I have not tested this, but wdyt about keeping the old variable, warning on its use , then removing it at a later date?